Library/Homebrew/extend/os/linux/development_tools.rb

```diff
@@ -15,5 +17,14 @@ def locate(tool)
     def default_compiler
       :gcc
     end
+
+    def build_system_info
+      build_info = {
+        "host_glibc_version" => OS::Linux::Glibc.system_version.to_s.presence,
+        "host_gcc_version"   => OS::Linux.host_gcc_version,
+        "kernel_headers"     => OS::Linux.kernel_headers_version,
```

Comment:
Let's 🔥 this for now as there hasn't been any pressing needs for this before this PR and it's all new code.